src/http_client.py

```diff
@@ -661,7 +670,16 @@ async def _refresh_credentials_base(self):
             if LOG_SENSITIVE_DATA:
                 log.debug("ROCKSTAR_REFRESH_JSON: " + str(final_json))
             filtered = final_request.cookies
-            new_auth = filtered['ScAuthTokenData2020'].value
+
+            auth_cookie = None
+            for cookie in filtered:
+                if cookie.find("TSc") != -1:
+                    auth_cookie = cookie
+                    break
+            else:
+                log.debug("ROCKSTAR_REFRESH_AUTH_COOKIE_FAILURE: The authentication cookie could not be found!")
```

Comment:
Realistically, it would have raised a `KeyError` in the following lines if the cookie didn't exist, which would also have been caught. Still, we may as well try and exit as soon as possible. Plus, your approach allows for customized exceptions, if they were so desired.

Reply:
Ah I didn't notice a big try-except above it, so the `KeyError` is handled, that's fine. Those methods is way too long for quick reading! 

I've also noticed that you're not using _get_bearer (and get _get_user_json) at all, maybe removing it would make the code more friendly for newcomers.